uaclient/cli.py

```diff
@@ -1004,6 +1035,68 @@ def action_attach(args, *, cfg):
     )
 
 
+def _write_command_output_to_file(
+    cmd, filename: str, return_codes: List[int] = None
+) -> None:
+    """Helper which runs a command and writes output or error to filename."""
+    try:
+        out, _ = util.subp(cmd.split(), rcs=return_codes)
+    except util.ProcessExecutionError as e:
+        util.write_file(filename, str(e))
+    else:
+        util.write_file(filename, out)
+
+
+def _collect_file(path: str, out_dir: str) -> None:
+    if os.path.isfile(path):
+        shutil.copy(path, out_dir)
+
+
+# We have to assert root here, because the logs are not non-root user readable
+@assert_root
+def action_collect_logs(args, *, cfg):
+    current_datetime = datetime.now().strftime(format="%b-%d-%H%M")
+
+    output_file = args.output or "ua_logs.tar.gz"
+    output_dir = "/tmp/ua-logs-{}".format(current_datetime)
+    if os.path.exists(output_dir):
+        shutil.rmtree(output_dir)
+    os.makedirs(output_dir)
+
+    _write_command_output_to_file(
+        "cloud-id", "{}/cloud-id.txt".format(output_dir)
+    )
+    _write_command_output_to_file(
+        "ua status --format json", "{}/ua-status.json".format(output_dir)
```

Comment:
Let's double check to see if there is no private information here

Reply:
Will do, and post the thoughts here

Reply:
Perfect, thanks for the explanation @blackboxsw